esphome/components/switch/__init__.py

```diff
@@ -56,6 +57,9 @@
 SwitchPublishAction = switch_ns.class_("SwitchPublishAction", automation.Action)
 
 SwitchCondition = switch_ns.class_("SwitchCondition", Condition)
+SwitchStateChangedTrigger = switch_ns.class_(
+    "SwitchStateChangedTrigger", automation.Trigger.template(bool)
+)
```

Comment:
Several other components use a shorter name, "StateTrigger". I'll leave it to you, but it might be nice to use that for consistency. 🙂
```suggestion
StateTrigger = switch_ns.class_(
    "StateTrigger", automation.Trigger.template(bool)
)
```

Reply:
I used that name to be consistent with `SwitchTurnOnTrigger` and `SwitchTurnOffTrigger` used on the same component. Probably even more consistent would be `SwitchStateTrigger`, which I would prefer. 😉

Please let me know your thoughts.